File: examples/verify-certificate.py
import sys
import json
from openexecution_verify import OpenExecutionVerifier


def main():
    if len(sys.argv) < 2:
        print("Usage: python verify-certificate.py <certificate-id>")
        sys.exit(1)

    verifier = OpenExecutionVerifier()
    try:
        result = verifier.verify_certificate(sys.argv[1])
        print(json.dumps(result, indent=2))
        print(
            "Certificate is VALID" if result.get("valid") else "Certificate is INVALID"
        )
    except Exception as e:
        print(f"Error: {e}")
        sys.exit(1)


if __name__ == "__main__":
    main()

File: examples/verify-certificate.js
const { OpenExecutionVerifier } = require('@openexecution/verify');

async function main() {
  const verifier = new OpenExecutionVerifier();
  const certificateId = process.argv[2];

  if (!certificateId) {
    console.error('Usage: node verify-certificate.js <certificate-id>');
    process.exit(1);
  }

  try {
    const result = await verifier.verifyCertificate(certificateId);
    console.log(JSON.stringify(result, null, 2));
    console.log(result.valid ? '\nCertificate is VALID' : '\nCertificate is INVALID');
  } catch (err) {
    console.error('Error:', err.message);
    process.exit(1);
  }
}

main();
